driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/internal/BatchCursorPublisherTest.java

```diff
@@ -116,6 +124,19 @@ public void testBatchCursorOnNextError() {
 
     }
 
+    @Test
+    public void testCancellingSubscriptionBatchCursor() {
+        List<Document> documents = IntStream.range(1, 11).boxed().map(i -> Document.parse(format("{_id: %s}", i))).collect(toList());
+
+        StepVerifier.create(createVerifiableBatchCursor(documents, 2), 1)
+                .expectNext(documents.get(0))
+                .thenRequest(2)
+                .expectNext(documents.get(1), documents.get(2))
+                .thenCancel()
```

Comment:
Tests cancelling the subscription and that the extra document from the batch cursor was discarded.  (The batch cursor is setup to return pairs of documents).

Reply:
These tests are so nice.